### PR TITLE
#1274 Add Date Selector to Date Delivered Modal

### LINKED ISSUE #1586: Finance Fixes - Add Date Selector to Date Delivered Modal
### Description

We want a user to be able to specify when a shipment for a reimbursement request was delivered. To facilitate this you'll need to update the backend endpoint to take a 'dateDelivered' Date, and add a field to the Date Delivered modal so that the user can specify a date.

### Acceptance Criteria

- [ ] The modal that pops up when you click 'Mark Delivered' from the dropdown on the individual reimbursement request page should have a field for a user to enter 'Date Delivered'
- [ ] The mark as delivered endpoint should be update to expect a payload of: dateDelivered: Date
- [ ] This date should be passed to the service method and used when updating the dateDelivered

### Proposed Solution
See the 'Report Refund' modal for an example of how to do the date input.

### PR DESCRIPTION
**Pull request by ryanhowe28**
## Changes

draft
## Notes

dont look at me

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1586

### CODE REVIEW

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 1]
```diff
@@ -643,7 +645,7 @@ export default class ReimbursementRequestService {
    * @throws AccessDeniedException if the creator of the request is not the submitter
    * @returns the updated reimbursement request
    */
-  static async markReimbursementRequestAsDelivered(submitter: User, reimbursementRequestId: string) {
+  static async markReimbursementRequestAsDelivered(submitter: User, reimbursementRequestId: string, dateMarkedDelivered: string) {
```
Comment (RChandler234) [inline src/backend/src/services/reimbursement-requests.services.ts:648]: just name this dateDelivered

File: src/frontend/src/hooks/finance.hooks.ts [hunk 2]
```diff
@@ -235,6 +235,17 @@ export const useMarkReimbursementRequestAsDelivered = (id: string) => {
   );
 };
 
+// export const useEditReimbursementRequest = (reimbursementRequestId: string) => {
```
Comment (RChandler234) [inline src/frontend/src/hooks/finance.hooks.ts:238]: don't comment this out

File: src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx [hunk 3]
```diff
@@ -151,17 +188,49 @@ const ReimbursementRequestDetailsView: React.FC<ReimbursementRequestDetailsViewP
     );
   };
 
+
   const MarkDeliveredModal = () => (
-    <NERModal
+    // <NERModal
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx:193]: delete

File: src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx [hunk 4]
```diff
@@ -99,17 +122,31 @@ const ReimbursementRequestDetailsView: React.FC<ReimbursementRequestDetailsViewP
...
-        toast.error(e.message, 3000);
+      toast.success('New Account Credit Reported Successfully');
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        toast.error(error.message);
       }
     }
   };
 
+    // const handleMarkDelivered = () => {
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx:139]: delete

File: src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx [hunk 5]
```diff
@@ -99,17 +122,31 @@ const ReimbursementRequestDetailsView: React.FC<ReimbursementRequestDetailsViewP
     }
   };
 
-  const handleMarkDelivered = () => {
+  const handleMarkDelivered = async (data: { dateDelivered: Date }) => {
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx:125]: this should be dateReceived, not dateDelivered that's why it's not submitting

File: src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx [hunk 6]
```diff
@@ -151,17 +188,49 @@ const ReimbursementRequestDetailsView: React.FC<ReimbursementRequestDetailsViewP
...
+
+
+    <NERFormModal
       open={showMarkDelivered}
       onHide={() => setShowMarkDelivered(false)}
       title="Warning!"
       cancelText="No"
       submitText="Yes"
-      onSubmit={handleMarkDelivered}
+      reset={reset}
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx:211]: this should be this `() => reset({ dateReceived: undefined })`

File: src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx [hunk 7]
```diff
@@ -77,6 +88,18 @@ const ReimbursementRequestDetailsView: React.FC<ReimbursementRequestDetailsViewP
   );
   const isSaboSubmitted = isReimbursementRequestSaboSubmitted(reimbursementRequest);
 
+  const {
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx:91]: I would move all this form code and the modal into it's own file like we do for: `src\frontend\src\pages\FinancePage\FinanceComponents\ReportRefundModal.tsx`

**Review by RChandler234 — CHANGES_REQUESTED** [review]
You'll need to update the backend tests because you updated the service method

**Comment by ryanhowe28** [comment]
Not close to being done

**Pull request closed without merging**